json_annotation/lib/src/allowed_keys_helpers.dart

```diff
@@ -2,16 +2,38 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// Helper function used in generated code when
+/// **DEPRECATED** Helper function used in generated code when
 /// `JsonSerializable.disallowUnrecognizedKeys` is `true`.
 ///
 /// Should not be used directly.
+@Deprecated('Code generated with the latest `json_serializable` will use '
+    '`\$checkKeys` instead. This function will be removed in the next majoy '
+    'release.')
 void $checkAllowedKeys(Map map, Iterable<String> allowedKeys) {
+  $checkKeys(map, allowedKeys: allowedKeys);
+}
+
+/// Helper function used in generated code when
+/// `JsonSerializable.disallowUnrecognizedKeys` is `true`.
+///
+/// Should not be used directly.
+void $checkKeys(Map map,
+    {Iterable<String> allowedKeys, List<String> requiredKeys}) {
   if (map == null) return;
```

Comment:
Should this still happen if `requiredKeys` is not empty?

Reply:
ack – although look at my fix